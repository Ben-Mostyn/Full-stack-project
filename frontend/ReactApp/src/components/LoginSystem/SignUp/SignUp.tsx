import { useMutation } from "@apollo/client";
import { CREATE_USER_MUTATION } from "../../../Graphql/create-user";
import { useState } from "react";

const SignUp = () => {

    const [createUser, { data, loading, error }] = useMutation(CREATE_USER_MUTATION);
    const [formData, setFormData] = useState({ username: '', email: '', password: '' });

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setFormData({ ...formData, [e.target.name]: e.target.value });
    };

    const handleSubmit = async (e: React.FormEvent) => {
        e.preventDefault();
        try {
            await createUser({ variables: formData });
            alert('User created successfully');
            setFormData({ username: '', email: '', password: '' })
        } catch (error) {
            console.error(error);
            alert('User creation failed')
        }
    }
    return (
        <div className="formContainer">
            <h2>Sign Up</h2>
            {loading && <p>Loading...</p>}
            {error && <p >Error: {error.message}</p>}
            {data && <p >User {data.createUser.username} created successfully!</p>}
            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    placeholder="username"
                    name="username" value={formData.username}
                    onChange={handleChange}
                    required
                />

                <input
                    type="text"
                    placeholder="email"
                    name='email'
                    onChange={handleChange}
                    value={formData.email}
                    required
                />

                <input
                    type="text"
                    placeholder="password"
                    name='password'
                    onChange={handleChange}
                    value={formData.password}
                    required
                />
                <button type="submit">Sign Up</button>
            </form>
        </div>
    );
}

export default SignUp;