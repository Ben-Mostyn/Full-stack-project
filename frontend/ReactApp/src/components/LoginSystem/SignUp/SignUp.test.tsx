import { render, screen } from '@testing-library/react';
import { describe, expect, test } from "vitest";
import SignUp from './SignUp';
import { MockedProvider } from '@apollo/client/testing';


describe('Sign up', () => {
    test('', () => {
        const {container} = render(
            <MockedProvider>
                <SignUp />
            </MockedProvider>
    
    );

    const heading = screen.getByRole('heading', {level: 2});
        expect(heading).toHaveTextContent('Sign Up');

        expect(container).toMatchSnapshot();
    })
})