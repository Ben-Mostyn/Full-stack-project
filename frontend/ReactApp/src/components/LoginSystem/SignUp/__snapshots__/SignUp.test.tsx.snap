// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sign up >  1`] = `
<div>
  <div
    class="formContainer"
  >
    <h2>
      Sign Up
    </h2>
    <form>
      <input
        name="username"
        placeholder="username"
        required=""
        type="text"
        value=""
      />
      <input
        name="email"
        placeholder="email"
        required=""
        type="text"
        value=""
      />
      <input
        name="password"
        placeholder="password"
        required=""
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;
